sdk-estimate-gas/README.md

```diff
@@ -0,0 +1,361 @@
+# Estimate the costs of an Optimistic (L2) transaction
+
+[![Discord](https://img.shields.io/discord/667044843901681675.svg?color=768AD4&label=discord&logo=https%3A%2F%2Fdiscordapp.com%2Fassets%2F8c9701b98ad4372b58f13fd9f65f966e.svg)](https://discord.com/channels/667044843901681675)
+[![Twitter Follow](https://img.shields.io/twitter/follow/optimismPBC.svg?label=optimismPBC&style=social)](https://twitter.com/optimismPBC)
+
+
+This tutorial teaches you how to use the Optimism SDK to estimate the gas costs of L2 transactions. 
+This calculation is complicated by the fact that the major cost is the cost of writing the transaction on L1, it doesn't work to just multiply the gas used by the transaction by the gas price, the same way you would on L1. 
+[You can read the details of the calculation here](https://help.optimism.io/hc/en-us/articles/4411895794715-Transaction-fees).
+
+
+
+## Prerequisites
+
+[The node script](./gas.js) makes these assumptions:
+
+1. You have [Node.js](https://nodejs.org/en/) running on your computer, as well as [yarn](https://classic.yarnpkg.com/lang/en/).
+1. There is network connectivity to a provider on the Optimistic Kovan L2 network, and to the npm package registry.
+1. The account it uses, [0xf5a6ead936fb47f342bb63e676479bddf26ebe1d](https://kovan-optimistic.etherscan.io/address/0xf5a6ead936fb47f342bb63e676479bddf26ebe1d), has enough Kovan ETH to pay for gas.
+   If it does not have enough Kovan ETH, please replenish it [from this faucet](https://faucet.paradigm.xyz/).
```

Comment:
Can alternatively link to the faucets page ("... please replenish it from once of the faucets found here"). We will have more testnet faucets soon.